benchmarks/distributed/zero/zero.py

```diff
@@ -0,0 +1,289 @@
+            torchvision.models.resnet50(pretrained=False),
+            resnet_output,
+            torch.randn(BATCH_SIZE, 1000),
+            nn.MSELoss(),
+            False,
+        ), (
+            "ResNet152",
+            torchvision.models.resnet152(pretrained=False),
+            resnet_output,
+            torch.randn(BATCH_SIZE, 1000),
+            nn.MSELoss(),
+            False,
+        ), (
+            "BERT",
+            transformers.BertModel.from_pretrained("bert-base-uncased"),
+            bert_output,
+            torch.randint(2, (BATCH_SIZE, 768)),
+            nn.CrossEntropyLoss(),
+            True,
+        ),
+    ]
+
+
+def dump_experiment_info(world_sizes, models, outdir):
+    r"""
+    Dumps the experiment info to a .json file named "exp_info.json".
+
+    Arguments:
+        world_sizes (List[int]): list of world sizes.
+        models (List[(str, int * DDP -> tensor, Tensor, _Loss, bool)]): a
+            list of models as described in `get_benchmark_models()`.
+        outdir (str): dirname to dump to.
+    """
+    experiment_info = dict()
+    experiment_info["world_sizes"] = world_sizes
+    experiment_info["model_names"] = [model_entry[0] for model_entry in models]
+    outfile = os.path.join(outdir, "exp_info.json")
+    with open(outfile, 'w') as f:
+        json.dump(experiment_info, f)
+
+
+# Worker Helpers #
+def get_parameter_sizes(rank, partitions):
+    r"""
+    Gets a list of parameter sizes for the given rank.
+
+    Arguments:
+        rank (int): rank to get parameter sizes for.
+        partitions (List[List[dict]]): parameter partitions returned by
+            partition_parameters().
+    """
+    sizes = list()
+    param_groups = partitions[rank]
+    for param_group in param_groups:
+        sizes.extend([param.numel() for param in param_group["params"]])
+    return sizes
+
+
+def worker_setup(rank, backend, world_size):
+    r"""
+    Sets up the environment and initializes the process group.
+
+    Arguments:
+        rank (int): process rank.
+        backend (str): name of the backend (e.g. "nccl", "gloo", etc.).
+        world_size (int): world size.
+    """
+    torch.manual_seed(0)
+    torch.cuda.manual_seed(0)
+    os.environ['MASTER_ADDR'] = 'localhost'
+    os.environ['MASTER_PORT'] = '29500'
+    dist.init_process_group(backend=backend, rank=rank, world_size=world_size)
+
+
+# Worker #
+def worker(rank, backend, world_size, model_name, model, outputs_fn, labels,
+           loss_fn, find_unused_parameters, outdir):
+    r"""
+    Finds the number of parameters in this process's partition of the
+    optimizer state and times the optimizer.step() latency.
+
+    Arguments:
+        rank (int): process rank.
+        backend (str): name of the backend (e.g. "nccl", "gloo", etc.).
+        world_size (int): world size.
+        model_name (str): human-readable name of the model.
+        output_fn (int * DDP -> Tensor): function mapping a (rank, ddp_model)
```

Comment:
the list here is in consistent with the real arg list, e.g., model is not in the doc, and `output_fn` -> `outputs_fn`.